docs/how_to/add_op_in_backend.md

```diff
@@ -0,0 +1,528 @@
+of the input tensor will not be overwritten by the output.
+- Line 20: Define the input argument name as `data` for the operator.
+- Line 21: Add user input parameters `a`, `b`, and `c` as the attributes of the operator.
+- Line 22: Register an operator named `_backward_quadratic` for backward pass
+of the operator `quadratic`. The underscore prefix in the operator name indicates
+that this is an operator not exposed to frontend and only used in backend.
+- Line 23: Set the parameter parser for the operator `_backward_quadratic`.
+- Line 24: Set the number of inputs.
+- Line 25: Set the number of outputs.
+- Line 26: Add `TIsBackward` attribute for the operator. The shape and type
+inference passes use this attribute to determine whether a node in the graph is a
+forward or backward node.
+- Line 27: Register backward function.
+
+So far, we have acquired an operator working on CPU in frontend.
+In order to register the operator working on GPUs, we just need to add the following
+code to `quadratic_op.cu`. Note that forward and backward functions
+are registered with attribute key `FCompute<gpu>`, rather than `FCompute<cpu>`.
+```cpp
+NNVM_REGISTER_OP(quadratic)
+.set_attr<FCompute>("FCompute<gpu>", QuadraticOpForward<gpu>);
+
+NNVM_REGISTER_OP(_backward_quadratic)
+.set_attr<FCompute>("FCompute<gpu>", QuadraticOpBackward<gpu>);
+```
+
+### Unit Test
+Now we have finished implementing the operator `quadratic` in MXNet backend.
+In order to unit test it in frontend, we need to add the following code
+to the python file `test_operator.py`. Note that while testing the
+forward pass is straightforward using `mx.nd.quadratic`, testing
+the backward involves a bit of more efforts. We create a
+`quadratic` symbol and feed it into the utility function `check_numeric_gradient`.
+The utility function will perform a perturbation on the input
+and calculate the response rate of the output using the
+[finite difference method](https://en.wikipedia.org/wiki/Finite_difference_method).
+Then it will compare the gradient from the backward pass with
+the values from the finite difference method. The test
+will be successful once the comparison satisfies user specified
+relative and absolute thresholds.
+```python
+def test_quadratic_function():
+    def f(x, a, b, c):
+        return a * x**2 + b * x + c
+
+    a = np.random.random_sample()
+    b = np.random.random_sample()
+    c = np.random.random_sample()
+    # check forward
+    for ndim in range(1, 6):
+        shape = rand_shape_nd(ndim, 5)
+        data = rand_ndarray(shape=shape, stype='default')
+        data_np = data.asnumpy()
+        expected = f(data_np, a, b, c)
+        output = mx.nd.quadratic(data, a=a, b=b, c=c)
+        assert_almost_equal(output.asnumpy(), expected)
+
+        # check backward using finite difference
+        data = mx.sym.Variable('data')
+        quad_sym = mx.sym.quadratic(data=data, a=a, b=b, c=c)
+        check_numeric_gradient(quad_sym, [data_np])
```

Comment:
Should we also mention `check_symbolic_forward` and `check_symbolic_backward` since they're usually used to test, too?

Reply:
Done.